rust/lance/src/index/vector/utils.rs

```diff
@@ -65,3 +70,53 @@ pub async fn maybe_sample_training_data(
     })?;
     Ok(array.as_fixed_size_list().clone())
 }
+
+#[derive(Debug)]
+pub struct PartitionLoadLock {
+    partition_locks: RwLock<HashMap<String, Arc<Mutex<()>>>>,
```

Comment:
if we make this `vec![Arc<Mutex()]` with pre-allocated mutex, we dont need `RWlock`?

Reply:
fixed